safety_limiter/src/safety_limiter.cpp

```diff
@@ -439,24 +442,40 @@ class SafetyLimiterNode
     if (!has_collision)
       return 1.0;
 
-    d_col = std::max<float>(std::abs(d_col) - d_margin_, 0.0);
-    yaw_col = std::max<float>(std::abs(yaw_col) - yaw_margin_, 0.0);
+    // delay compensation:
+    //   d_compensated = d - delay * sqrt(2 * acc * |d_compensated|)
+    //   d_compensated = d + acc * delay^2 - sqrt((acc * delay^2)^2 + 2 * d * acc * delay^2)
```

Comment:
We can obtain 4 results by solving the first equation by Wolfram alpha.
Your solution is the third one in the below solutions. We can ignore other solutions?

![solve x x=p_1 - p_2 - sqrt(2-p_3-abs(x)) - Wolfram|Alpha - Google Chrome_355](https://user-images.githubusercontent.com/6742467/59025533-58060080-888f-11e9-98ff-02fa26fab42f.png)

Reply:
In the actual calculation, sign of x is handled outside of this formula. Two of these four results are caused by entering both x and abs(x). These cases are not used.
Another one case should mean that the sign of both side of the first formula is different. This is also not a real condition. Negative value of distance is treated outside of this formula.

Reply:
I see, thanks for the detailed explanation. I found that additional restriction x >= 0 results in your solution seems correct  [here](https://www.wolframalpha.com/input/?i=solve+x%3E%3D0,+x%3Da+-+b+*+sqrt(2*c*x)). Will approve.